### PR TITLE
More concrete eslint rules

### LINKED ISSUE #525: Improve linter configuration
Currently we have quite exotic configuration which contains lots of legacy stuff. Let's try to clean this.

### PR DESCRIPTION
**Pull request by Nokel81**
Codify repo rules, since I don't think we check these on merge (yet) we can just discuss these here.

Once any changes have been approved. I will work (as quickly as possible) to make the non-auto-fixable fixes (with no other changes) and merge that.

fixes #525

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -3,37 +3,38 @@
   "productName": "Lens",
   "description": "Lens - The Kubernetes IDE",
   "version": "3.6.0-dev",
-  "main": "static/build/main.js",
-  "copyright": "© 2020, Mirantis, Inc.",
+  "main": "out/main.js",
```
Comment (ixrock) [inline package.json]: This is incorrect path to `main.js` in `master`. Wrong merge?

File: package.json [hunk 2]
```diff
@@ -3,37 +3,38 @@
...
+  "copyright": "© 2020, Lakend Labs, Inc.",
   "license": "MIT",
   "author": {
-    "name": "Mirantis, Inc.",
-    "email": "info@k8slens.dev"
+    "name": "Lakend Labs, Inc.",
+    "email": "info@lakendlabs.com"
```
Comment (ixrock) [inline package.json]: Here as well, should be `Mirantis, Inc.`

File: package.json [hunk 3]
```diff
@@ -3,37 +3,38 @@
...
-    "build:linux": "yarn compile && electron-builder --linux --dir -c.productName=Lens",
-    "build:mac": "yarn compile && electron-builder --mac --dir -c.productName=Lens",
-    "build:win": "yarn compile && electron-builder --win --dir -c.productName=Lens",
+    "compile": "concurrently \"yarn i18n:compile\" \"yarn compile:main -p\" \"yarn compile:renderer -p\"",
+    "compile:main": "webpack --progress --config webpack.main.ts",
+    "compile:renderer": "webpack --progress --config webpack.renderer.ts",
+    "compile:dll": "webpack --config webpack.dll.ts",
+    "build:linux": "yarn compile && electron-builder --linux --dir -c.productName=LensDev",
+    "build:mac": "yarn compile && electron-builder --mac --dir -c.productName=LensDev",
+    "build:win": "yarn compile && electron-builder --win --dir -c.productName=LensDev",
```
Comment (ixrock) [inline package.json]: So, looks like whole file quite outdated.

File: package.json (hunk 1 above)
Comment (Nokel81) [reply package.json]: Yes

File: package.json (hunk 2 above)
Comment (Nokel81) [reply package.json]: oops

File: package.json (hunk 3 above)
Comment (Nokel81) [reply package.json]: wow, okay. oops

**Comment by Nokel81** [comment]
@ixrock definitely a bad cherry-pick. Revert to master and did the `yarn add` directly

**Comment by Nokel81** [comment]
Closing due to the implicit size of changing all these rules at once.

**Pull request closed without merging**